### PR TITLE
HD-104389 Accounts > Confirm that oauth failure on accounts can be safely recovered/retried

### PR DESCRIPTION
**Pull request by annefrmarinas**
## Ticket references
- [HD-104389 Accounts > Confirm that oauth failure on accounts can be safely recovered/retried](https://freedom.myjetbrains.com/youtrack/issue/HD-104389)

## Summary of changes

Added retry on cuddle requests to accounts. Default max number of retries is 3, but it can be configured by other projects using `accounts.configure({..., retry_count: <value>, ...})`; 

Unit test has also been added for the retry functionality.

Package version was updated to 1.1.0

## Checklist

- [x] bugfix
- [ ] new feature
- [ ] breaking change
- [x] added unit tests
- [x] changes introduced has been discussed and approved
- [ ] requires manual testing
- [ ] documentation has been updated

## Testing

* Execute `grunt test` in root directory. Verify that all tests are passing. No manual tests needed since additional unit tests already covered it.

### CODE REVIEW

**Review by jkcdarunday — DISMISSED** [review]
Please update README.md to showcase the retry ability.

**Review by jkcdarunday — DISMISSED** [review]

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "freedom-accounts-util",
-  "version": "1.0.0",
+  "version": "1.0.1",
```
Comment (wildtron) [inline package.json]: should be 1.1.0.. since we've added a new option

**Review by wildtron — DISMISSED** [review]

File: package.json (hunk 1 above)
Comment (annefrmarinas) [reply package.json]: Done. PR desciption updated as well.

**Review by jkcdarunday — DISMISSED** [review]

File: README.md [hunk 2]
```diff
@@ -16,13 +16,14 @@ yarn add freedom-accounts-util
 ```
 const accounts = require('freedom-accounts-util');
 
-accounts.configure({base_url: '', path: '', client_id: '', client_secret: '', client_expiry: 300, server_expiry: 300, disable_caching: false});
+accounts.configure({base_url: '', path: '', client_id: '', client_secret: '', client_expiry: 300, server_expiry: 300, disable_caching: false, retry_count: ''});
```
Comment (sibayanjasper) [inline README.md]: prettify? hard to read the fields.
```js
accounts.configure({
    base_url: '',
    path: '',
    client_id: '',
    client_secret: '',
    client_expiry: 300,
    server_expiry: 300,
    disable_caching: false,
    retry_count: ''}
);
```
Also, retry_count should be number, and should have default value of 3 I guess?

**Review by sibayanjasper — DISMISSED** [review]

File: README.md (hunk 2 above)
Comment (sibayanjasper) [reply README.md]: And comment client_expiry, and server_expiry to be in seconds.

Comment (annefrmarinas) [reply README.md]: Prettified :D Comment for client_expiry and server_expiry to be in seconds is already included in the `Notes` section of README

**Review by jkcdarunday — APPROVED** [review]

**Review by wildtron — APPROVED** [review]

**Review by sibayanjasper — APPROVED** [review]

**Pull request merged**